qubes/ext/utils.py

```diff
@@ -100,3 +159,56 @@ def compare_device_cache(vm, devices_cache, current_devices):
             if cached_front is not None:
                 detached[dev_id] = cached_front
     return added, attached, detached, removed
+
+
+async def confirm_device_attachment(device, frontends) -> str:
+    try:
+        return await _do_confirm_device_attachment(device, frontends)
+    except Exception as exc:
+        print(str(exc.__class__.__name__) + ":", str(exc), file=sys.stderr)
+        return ""
+
+
+async def _do_confirm_device_attachment(device, frontends):
+    socket = "device-agent.GUI"
+
+    app = tuple(frontends.keys())[0].app
+    doms = app.domains
+
+    front_names = [f.name for f in frontends.keys()]
+
+    try:
+        guivm = doms["dom0"].guivm.name
+    except AttributeError:
+        guivm = "dom0"
+
+    number_of_targets = len(front_names)
+
+    params = {
+        "source": device.backend_domain.name,
+        "device_name": device.description,
+        "argument": device.port_id,
+        "targets": front_names,
+        "default_target": front_names[0] if number_of_targets == 1 else "",
+        "icons": {
+            (
+                dom.name if dom.klass != "DispVM" else f"@dispvm:{dom.name}"
```

Comment:
shouldn't that be `getattr(dom, "template_for_dispvms", False)` check instead? DispVM class is already running one, not a template to create new one from.

Reply:
Yes, it's for running DispVMs to indicate in the vm list which vms are disposible